Error while checking build/O2/o2-cs8 for 70fd71cfcfe5b96b8b8813ebf3b2a457faaf42ed at 2022-09-05 01:56:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 446
 88/106 Test #102: test_Framework_test_SimpleDataProcessingDevice01 ........***Failed    2.20 sec
[ERROR] pid 13569 (internal-dpl-injected-dummy-sink) crashed with 1
[ERROR] SEVERE: Device internal-dpl-injected-dummy-sink (13569) returned with 1
99% tests passed, 1 tests failed out of 101
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9746/70fd71cfcfe5b96b8b8813ebf3b2a457faaf42ed/build_O2_o2-cs8/pretty.html).

--- a/Detectors/Calibration/include/DetectorsCalibration/TimeSlot.h
+++ b/Detectors/Calibration/include/DetectorsCalibration/TimeSlot.h
@@ -34,10 +34,7 @@ class TimeSlot
 {
  public:
   TimeSlot() = default;
-  TimeSlot(TFType tfS, TFType tfE) : mTFStart(tfS), mTFEnd(tfE)
-  {
-    mTFStartMS = getStartTimeMS();
-  }
+  TimeSlot(TFType tfS, TFType tfE) : mTFStart(tfS), mTFEnd(tfE) {}
   TimeSlot(const TimeSlot& src) : mTFStart(src.mTFStart), mTFEnd(src.mTFEnd), mContainer(std::make_unique<Container>(*src.getContainer())) {}
   TimeSlot& operator=(const TimeSlot& src)
   {
@@ -65,6 +62,7 @@ class TimeSlot
 
   void setTFStart(TFType v) { mTFStart = v; }
   void setTFEnd(TFType v) { mTFEnd = v; }
+  void setStaticStartTimeMS(long t) { mTFStartMS = t; }
   void setRunStartOrbit(long t) { mRunStartOrbit = t; }
   auto getRunStartOrbit() const { return mRunStartOrbit; }
 
@@ -76,11 +74,12 @@ class TimeSlot
   {
     mContainer->merge(prev.mContainer.get());
     mTFStart = prev.mTFStart;
+    mTFStartMS = prev.mTFStartMS;
   }
 
   void print() const
   {
-    LOGF(info, "Calibration slot %5d <=TF<=  %5d", mTFStart, mTFEnd);
+    LOGF(info, "Calibration slot %5d <=TF<=  %5d (start in ms = %ld)", mTFStart, mTFEnd, mTFStartMS);
     mContainer->print();
   }
 

--- a/Detectors/Calibration/include/DetectorsCalibration/TimeSlotCalibration.h
+++ b/Detectors/Calibration/include/DetectorsCalibration/TimeSlotCalibration.h
@@ -70,7 +70,7 @@ class TimeSlotCalibration
   void checkSlotLength()
   {
     if (mSlotLengthInSeconds > 0) {
-      TFType ntf = mSlotLengthInSeconds / (long(o2::base::GRPGeomHelper::getNHBFPerTF()) * o2::constants::lhc::LHCOrbitMUS / 1000000);
+      TFType ntf = mSlotLengthInSeconds / (o2::base::GRPGeomHelper::getNHBFPerTF() * o2::constants::lhc::LHCOrbitMUS * 1e-6);
       LOGP(info, "Redefining slot duration from {} s. to {} TFs", mSlotLengthInSeconds, ntf);
       setSlotLength(ntf);
       mSlotLengthInSeconds = 0;
@@ -343,6 +343,7 @@ TimeSlot<Container>& TimeSlotCalibration<Input, Container>::getSlotForTF(TFType 
     } else if (mSlots.empty()) {
       auto& sl = emplaceNewSlot(true, mFirstTF, tf);
       sl.setRunStartOrbit(getRunStartOrbit());
+      sl.setStaticStartTimeMS(sl.getStartTimeMS());
     }
     return mSlots.back();
   }
@@ -356,6 +357,7 @@ TimeSlot<Container>& TimeSlotCalibration<Input, Container>::getSlotForTF(TFType 
       LOG(info) << "Adding new slot for " << tfmn << " <= TF <= " << tfmx;
       auto& sl = emplaceNewSlot(true, tfmn, tfmx);
       sl.setRunStartOrbit(getRunStartOrbit());
+      sl.setStaticStartTimeMS(sl.getStartTimeMS());
       if (!tfmn) {
         break;
       }
@@ -377,6 +379,7 @@ TimeSlot<Container>& TimeSlotCalibration<Input, Container>::getSlotForTF(TFType 
     LOG(info) << "Adding new slot for " << tfmn << " <= TF <= " << tfmx;
     auto& sl = emplaceNewSlot(false, tfmn, tfmx);
     sl.setRunStartOrbit(getRunStartOrbit());
+    sl.setStaticStartTimeMS(sl.getStartTimeMS());
     tfmn = tf2SlotMin(mSlots.back().getTFEnd() + 1);
   } while (tf > mSlots.back().getTFEnd());
 

--- a/Detectors/TOF/calibration/src/LHCClockCalibrator.cxx
+++ b/Detectors/TOF/calibration/src/LHCClockCalibrator.cxx
@@ -88,6 +88,10 @@ void LHCClockCalibrator::initOutput()
 void LHCClockCalibrator::finalizeSlot(Slot& slot)
 {
   // Extract results for the single slot
+
+  // marging in milliseconds for the end validity of the object to be uploaded
+  static long endValidityMarging = long((5 + getMaxSlotsDelay()) * getSlotLength() * o2::base::GRPGeomHelper::getNHBFPerTF() * o2::constants::lhc::LHCOrbitMUS * 1e-3);
+
   o2::tof::LHCClockDataHisto* c = slot.getContainer();
   LOG(info) << "Finalize slot " << slot.getTFStart() << " <= TF <= " << slot.getTFEnd() << " with "
             << c->getEntries() << " entries";
@@ -108,8 +112,8 @@ void LHCClockCalibrator::finalizeSlot(Slot& slot)
   auto clName = o2::utils::MemFileHelper::getClassName(l);
   auto flName = o2::ccdb::CcdbApi::generateFileName(clName);
 
-  auto starting = slot.getStartTimeMS();
-  auto stopping = slot.getEndTimeMS() + 5 * getSlotLength() + getMaxSlotsDelay();
+  auto starting = slot.getStartTimeMS() - o2::ccdb::CcdbObjectInfo::SECOND * 10; // adding a marging, in case some TFs were not processed
+  auto stopping = slot.getEndTimeMS() + endValidityMarging;
   LOG(info) << "starting = " << starting << " - stopping = " << stopping << " -> phase = " << fitValues[1] << " ps (added BC = " << tobeused << ")";
   l.setStartValidity(starting);
   l.setEndValidity(stopping);

--- a/Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx
+++ b/Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx
@@ -627,7 +627,7 @@ void TOFChannelCalibrator<T>::finalizeSlotWithCosmics(Slot& slot)
 
   auto clName = o2::utils::MemFileHelper::getClassName(ts);
   auto flName = o2::ccdb::CcdbApi::generateFileName(clName);
-  auto startValidity = slot.getStartTimeMS();
+  auto startValidity = slot.getStaticStartTimeMS() - o2::ccdb::CcdbObjectInfo::SECOND * 10; // adding a marging, in case some TFs were not processed
   auto endValidity = o2::ccdb::CcdbObjectInfo::MONTH * 2;
   ts.setStartValidity(startValidity);
   ts.setEndValidity(endValidity);
@@ -734,7 +734,7 @@ void TOFChannelCalibrator<T>::finalizeSlotWithTracks(Slot& slot)
 
       double fitres = fitGaus(nbinsUsed, histoValues.data(), minRange, maxRange, fitValues, nullptr, 2., false);
       LOG(info) << "channel = " << ich << " fitted by thread = " << ithread;
-      if (fitres > -3) {
+      if (fitres > -10) {
         LOG(info) << "Channel " << ich << " :: Fit result " << fitres << " Mean = " << fitValues[1] << " Sigma = " << fitValues[2];
       } else {
 #ifdef DEBUGGING
@@ -801,7 +801,7 @@ void TOFChannelCalibrator<T>::finalizeSlotWithTracks(Slot& slot)
   }   // end loop over sectors
   auto clName = o2::utils::MemFileHelper::getClassName(ts);
   auto flName = o2::ccdb::CcdbApi::generateFileName(clName);
-  auto startValidity = slot.getStaticStartTimeMS();
+  auto startValidity = slot.getStaticStartTimeMS() - o2::ccdb::CcdbObjectInfo::SECOND * 10; // adding a marging, in case some TFs were not processed
   auto endValidity = startValidity + o2::ccdb::CcdbObjectInfo::MONTH * 2;
   ts.setStartValidity(startValidity);
   ts.setEndValidity(endValidity);

